### PR TITLE
ddl:add gofail test in ddl

### PR DESCRIPTION
**Pull request by ciscoxll**
### What problem does this PR solve? 

add gofail test cases in unit test.

### What is changed and how it works?
```
// enable gofail
gofail.Enable("github.com/pingcap/tidb/ddl/truncateTableErr", `return(true)`)
// disable gofail
defer gofail.Disable("github.com/pingcap/tidb/ddl/truncateTableErr")
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: ddl/column.go [hunk 1]
```diff
@@ -230,6 +230,12 @@ func onDropColumn(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 		return ver, errors.Trace(err)
 	}
 
+	// gofail: var dropColumnErr bool
+	// if dropColumnErr {
+	// job.State = model.JobStateCancelled
```
Comment (winkyao) [inline ddl/column.go]: Why you set model.JobStateCancelled here？

File: ddl/schema.go [hunk 2]
```diff
@@ -76,6 +76,12 @@ func onDropSchema(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 		return ver, errors.Trace(err)
 	}
 
+	// gofail: var dropSchemaErr bool
+	// if dropSchemaErr {
+	// job.State = model.JobStateCancelled
```
Comment (winkyao) [inline ddl/schema.go]: ditto

File: ddl/table.go [hunk 3]
```diff
@@ -127,6 +127,12 @@ func onDropTableOrView(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 		return ver, errors.Trace(err)
 	}
 
+	// gofail: var dropTableOrViewErr bool
+	// if dropTableOrViewErr {
+	// job.State = model.JobStateCancelled
```
Comment (winkyao) [inline ddl/table.go]: ditto

**Comment by winkyao** [comment]
Please add some gofail in the key path of the ddl job, like addPhysicalTableIndex, splitTableRanges, backfillIndexInTxn, etc.

**Comment by ciscoxll** [comment]
@crazycs520 PTAL.

**Pull request closed without merging**